lesson17/script.js

```diff
@@ -0,0 +1,123 @@
+const slider = document.getElementById("js-slider");
+
+const createAttributedElements = ({tag,attrObj,str}) =>{
+    const element = document.createElement(tag);
+    Object.keys(attrObj).forEach((attribute) => {
+        element.setAttribute(attribute, attrObj[attribute]);
+    });
+    if(str !== undefined) element.textContent = str;
+    return element;
+}
+
+const createErrorMessage = (error) => {
+    const errMsg = createAttributedElements({tag:"p",attrObj:{class:"error-msg"},str:error})
+
+    slider.appendChild(errMsg);
+}
+
+const showLoading = () => {
+    const loadingImg = createAttributedElements({tag:"img",
+        attrObj:{
+        id:"js-loading",
+        class:"loading",
+        src:"img/loading-circle.gif",
+        alt:"ローディング"
+        }
+    });
+    slider.appendChild(loadingImg);
+}
+
+const removeLoading = () => document.getElementById("js-loading").remove();
+
+const fetchData = async (URL) => {
+    const response = await fetch(URL);
+    if(!response.ok){
+        console.error(`${response.status}:${response.statusText}`);
+        createErrorMessage('現在、サーバーの通信が壊れています');
+    }
+    const json = await response.json();
+    const data = json.data;
+    return data;
+}
+
+const fetchSliderData = (ms) => new Promise(resolve => setTimeout(() => resolve(fetchData("https://myjson.dit.upm.es/api/bins/2bzx")),ms));
+
+const init = async () => {
+    showLoading();
+    try{
+        const sliderData = await fetchSliderData(3000);
+        if (!sliderData) {
+            return;
+        }
+        renderSliderItems(sliderData);
+        getPageNumber(sliderData);
+        
+        renderButtons();
+        isActiveFirstSliderItem();
+        switchSliderItem();
+    }catch(error){
+        console.error(error);
+        createErrorMessage(error);
+    }finally{
+        removeLoading();
+    }
+}
+
+const renderSliderItems = (sliderData) => {
+    const fragment = document.createDocumentFragment();
+    for(const image of sliderData){
+        const sliderItem = createAttributedElements({tag:"div",attrObj:{class:"slider__item"}});
+        const slideImage = createAttributedElements({tag:"img",attrObj:{class:"slide-image",src:image.img,alt:image.alt}});
+        
+        fragment.appendChild(sliderItem).appendChild(slideImage);
+    }
+    slider.appendChild(fragment);
+}
+
+const renderButtons = () => {
+    const prevBtn = createAttributedElements({tag:"button",attrObj:{id:"js-prevBtn",class:"js-btn prev-btn slider-btn"}});
+    const nextBtn = createAttributedElements({tag:"button",attrObj:{id:"js-nextBtn",class:"js-btn next-btn slider-btn"}});
+    prevBtn.disabled = "true";
+
+    slider.appendChild(prevBtn).insertAdjacentElement("afterend",nextBtn);
+}
```

Comment:
IMO
`disabled: true`
We can put this in too, right?
```suggestion
const renderButtons = () => {
  const prevBtn = createAttributedElements({
    tag: "button",
    attrObj: {
      id: "js-prevBtn",
      class: "js-btn prev-btn slider-btn",
      disabled: true
    }
  });
  const nextBtn = createAttributedElements({
    tag: "button",
    attrObj: { id: "js-nextBtn", class: "js-btn next-btn slider-btn" }
  });

  slider.appendChild(prevBtn).insertAdjacentElement("afterend", nextBtn);
};
```

Reply:
Yes,I think so you are right.
I fixed.
https://github.com/nakagawamai/js-lesson/pull/19/commits/3ec178f03c2391ca524838d7110e6a5e0f1721a2

Reply:
Thank you for telling me something very important!
I wasn't aware of the value type.

I fixed.
https://github.com/nakagawamai/js-lesson/pull/19/commits/ced2543371ee4c51390efe7c7ee44aef10e7973e